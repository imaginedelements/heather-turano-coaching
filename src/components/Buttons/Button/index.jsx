import React from "react";
import PropTypes from "prop-types";

import { Icon, Paragraph } from "../../typography";

import "./index.module.scss";

export const Button = ({
  label,
  styleType,
  htmlType,
  disabled,
  loading,
  onClick,
  onSubmit
}) => (
  /* eslint-disable react/button-has-type */
  <button
    styleName={styleType}
    type={htmlType}
    disabled={disabled || loading}
    onClick={onClick}
    onSubmit={onSubmit}
  >
    {loading ? (
      <Icon size="md" copy={label} icon="spinner" spin />
    ) : (
      <Paragraph size="md">{label}</Paragraph>
    )}
  </button>
  /* eslint-enable react/button-has-type */
);

Button.propTypes = {
  /**
   * the button text
   */
  label: PropTypes.string.isRequired,
  /**
   * the type of button style that should be applied
   */
  styleType: PropTypes.oneOf([
    "primary",
    "secondary",
    "alt",
    "small",
    "destructive"
  ]),
  /**
   * the htmlType of the button
   */
  htmlType: PropTypes.oneOf(["button", "submit", "reset"]),

  /**
   * the action that is fired after the user clicks the button
   */
  onClick: PropTypes.func,
  /**
   * the action that is fired after the user submits the form
   */
  onSubmit: PropTypes.func,
  /**
   * a truthy value disables the button and the user can no longer click on it
   */
  disabled: PropTypes.bool,
  /**
   * a truthy value disables the button and displays a loading spinner.
   * If a button is loading, then it is also disabled as well
   */
  loading: PropTypes.bool
};

Button.defaultProps = {
  styleType: "primary",
  htmlType: "button",
  disabled: false,
  onClick: null,
  onSubmit: null,
  loading: false
};

export default Button;
