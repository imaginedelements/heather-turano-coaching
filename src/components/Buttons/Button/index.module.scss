@import "../index";

button {
  &.primary {
    @include button--primary;
  }

  &.secondary {
    @include button--secondary;
  }

  &.destructive {
    @include button--destructive;
  }
}
