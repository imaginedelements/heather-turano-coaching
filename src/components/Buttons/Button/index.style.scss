@import "~@bit/cadent.styles.sass.buttons";

button {
  &.primary {
    @include button--primary;
  }

  &.secondary {
    @include button--secondary;
  }

  &.alt {
    @include button--alt;
  }

  &.small {
    @include button--small;
  }

  &.destructive {
    @include button--destructive;
  }
}
