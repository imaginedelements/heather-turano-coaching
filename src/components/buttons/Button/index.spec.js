import Button from "./index";

const props = {
  styleName: "primary",
  disabled: false,
  kind: "primary",
  action: null,
  label: "i am a button"
};

describe("<Button />", () => {
  it("should render correctly", () => {
    const wrapper = shallowWrapper(Button)(props);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render as a button with a type of button", () => {
    const wrapper = shallowWrapper(Button)({ ...props, type: "button" });
    expect(wrapper.type()).toEqual("button");
    expect(wrapper.find('[type="button"]')).toHaveLength(1);
  });

  it("should render as a button with a type of submit", () => {
    const wrapper = shallowWrapper(Button)({ ...props, type: "submit" });
    expect(wrapper.find('[type="submit"]')).toHaveLength(1);
  });
});
