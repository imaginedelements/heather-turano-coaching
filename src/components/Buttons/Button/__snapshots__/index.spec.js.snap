// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<button
  className="primary"
  disabled={false}
  onClick={null}
  onSubmit={null}
  type="button"
>
  <Label
    copy={null}
    size="lg"
  >
    i am a button
  </Label>
</button>
`;
